include/onnxruntime/core/providers/dml/dml_provider_factory.h

```diff
@@ -40,10 +44,58 @@ ORT_API_STATUS(OrtSessionOptionsAppendExecutionProvider_DML, _In_ OrtSessionOpti
  * objects.
  * See also: DMLCreateDevice
  * See also: ID3D12Device::CreateCommandQueue
+ * 
+ * [[deprecated]]
+ * This export should be deprecated.
+ * The OrtSessionOptionsAppendExecutionProvider_DML1 export on the OrtDmlApi should be used instead.
  */
 ORT_API_STATUS(OrtSessionOptionsAppendExecutionProviderEx_DML, _In_ OrtSessionOptions* options,
                _In_ IDMLDevice* dml_device, _In_ ID3D12CommandQueue* cmd_queue);
 
+
+struct OrtDmlApi;
+typedef struct OrtDmlApi OrtDmlApi;
+
+struct OrtDmlApi {
+  /**
+   * Creates a DirectML Execution Provider which executes on the hardware adapter with the given device_id, also known as
+   * the adapter index. The device ID corresponds to the enumeration order of hardware adapters as given by 
+   * IDXGIFactory::EnumAdapters. A device_id of 0 always corresponds to the default adapter, which is typically the 
+   * primary display GPU installed on the system. A negative device_id is invalid.
+  */
+  ORT_API2_STATUS(OrtSessionOptionsAppendExecutionProvider_DML, _In_ OrtSessionOptions* options, int device_id);
+
+  /**
+   * Creates a DirectML Execution Provider using the given DirectML device, and which executes work on the supplied D3D12
+   * command queue. The DirectML device and D3D12 command queue must have the same parent ID3D12Device, or an error will
+   * be returned. The D3D12 command queue must be of type DIRECT or COMPUTE (see D3D12_COMMAND_LIST_TYPE). If this 
+   * function succeeds, the inference session maintains a strong reference on both the dml_device and the command_queue 
+   * objects.
+   * See also: DMLCreateDevice
+   * See also: ID3D12Device::CreateCommandQueue
+   */
+  ORT_API2_STATUS(OrtSessionOptionsAppendExecutionProvider_DML1, _In_ OrtSessionOptions* options,
+                _In_ IDMLDevice* dml_device, _In_ ID3D12CommandQueue* cmd_queue);
+
+  /**
+    * DmlCreateGPUAllocationFromD3DResource
+    * This api is used to create a DML EP input based on a user specified d3d12 resource.
+  */
+  ORT_API2_STATUS(DmlCreateGPUAllocationFromD3DResource, _In_ ID3D12Resource* d3d_resource, _Out_ void** dml_resource);
```

Comment:
```suggestion
  ORT_API2_STATUS(CreateGPUAllocationFromD3DResource, _In_ ID3D12Resource* d3d_resource, _Out_ void** dml_resource);
```

I'm assuming we don't plan to expose CUDA/OpenVino functionality on the the DML EP API :b, and so the `Dml` is redundant.